### PR TITLE
Fix PortHandler

### PR DESCRIPTION
**Pull request by haslinghuis**
@McGiverGim found an issue with some GUI function continuous being called.

Fixes and improves following issues:

- [x] GUI updates were done in PortHandler every 500ms, moving them where status needs change.
- [x] Isolated PortHandler change handler in firmware flasher tab as it was working on other tabs.
- [x] PortHandler was cutting of text and adjusted the correction.
- [x] Auto-detect button was not always updated correctly.
- [x] Add recognition for SPRacing boards.
- [x] Disable flash button when flashing.
- [x] Don't check serial devices in DFU mode.
- [x] Don't check USB devices in serial mode. 

Due to changes it was needed to rethink how to enable the auto-detect button.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/port_handler.js
Comment (McGiverGim) [inline src/js/port_handler.js:270]: I don't like too much to have exclusions for one vendor. I think SPR must include STM in the name. I suppose is a customized variant of STM right?

**Review by McGiverGim — DISMISSED** [review]
It's ok to me, only one thing I don't like, but approved.

File: src/js/port_handler.js
Comment (haslinghuis) [reply src/js/port_handler.js:270]: The STM32H750 betaflight (legacy) target is shown as `/dev/tty/ACM0 - SPRacingH7EXTREME`

File: src/js/port_handler.js
Comment (chmelevskij) [inline src/js/port_handler.js:27]: Didn't we move to sync api now? 
```suggestion
    self.showVirtualMode = ConfigStorage.get('showVirtualMode').showVirtualMode);
    self.showAllSerialDevices = ConfigStorage.get('showAllSerialDevices').showAllSerialDevices;
```

**Review by chmelevskij — DISMISSED** [review]

File: src/js/port_handler.js
Comment (haslinghuis) [reply src/js/port_handler.js:27]: Yes here: https://github.com/betaflight/betaflight-configurator/pull/2636 (will do afterwards when rebasing).

Comment (haslinghuis) [reply src/js/port_handler.js]: :partying_face: Finally got it approved. Rebased now.

File: src/js/port_handler.js
Comment (haslinghuis) [reply src/js/port_handler.js]: @McGiverGim have removed it, agree this should not be vendor specific, as the board is auto selected anyway.

**Review by McGiverGim — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Pull request merged**
